lib/mocha-tests-lint.ts

```diff
@@ -0,0 +1,34 @@
+import * as fs from 'fs';
+
+// Mocha tests linter results.
+// When isError is set to true, the message contains details about found errors.
+export class MochaListResult {
+	constructor(
+		public readonly message: string,
+		public readonly isError: boolean,
+	) {}
+}
+
+const checkPattern = /((describe)|(it))\.only/;
+
+// Check if files in the test directory don't contain describe.only or it.only statements
+// to ensure that an accidental commit does not prevent all the tests from running.
+export function lintMochaTests(scripts: string[]): MochaListResult {
+	let errorsFound = false;
+	let message = '';
+
+	for (const scriptPath of scripts) {
+		const content = fs.readFileSync(scriptPath, { encoding: 'utf8' });
```

Comment:
I'd rather this wasn't a synchronous file access.

Reply:
Would you elaborate on that? What potentials problems are thinking about?

Reply:
In general it blocks the event loop which means nothing else can happen at the same time, and in this case that'll mean it takes longer to lint compared to if the file reads were done in parallel (potentially significantly longer with lots of mocha files)

Reply:
Parallelized.